CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -263,7 +263,9 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
           Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$accountRelationshipID] = $incomeAccountFinancialAccountID;
         }
       }
-
+      if (!isset(Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId])) {
+        Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId] = NULL;
```

Comment:
@pradpnayak OK - I've backed it up & made the function return NULL if there is nothing found - that seems consistent with what the non-cached function does